storage/src/sparse_merkle/tree.cpp

```diff
@@ -0,0 +1,198 @@
+// Concord
+//
+// Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the
+// LICENSE file.
+
+#include "sparse_merkle/tree.h"
+#include "hash_defs.h"
+
+using namespace concordUtils;
+
+namespace concord::storage::sparse_merkle {
+
+// All data that is loaded from the DB, manipulated, and/or returned from a
+// call to `update`.
+//
+// A new instance of this structure is created during every update call.
+struct UpdateCache {
+  UpdateCache(const BatchedInternalNode& root, const std::shared_ptr<IDBReader>& db_reader) : db_reader_(db_reader) {
+    internal_nodes_[NibblePath()] = root;
+  }
+
+  // Get a node if it's in the cache, otherwise get it from the DB.
+  BatchedInternalNode getInternalNode(const InternalNodeKey& key) {
+    auto it = internal_nodes_.find(key.path());
```

Comment:
In this MR you sometimes use struct members with trailing '_' and sometimes not :-)